packages/graphql-hooks/src/useClientRequest.ts

```diff
@@ -87,18 +90,15 @@ function useClientRequest<
   Variables = object,
   TGraphQLError extends GraphQLResponseError = GraphQLResponseError
 >(
-  query: string,
+  query: string | TypedDocumentNode<ResponseData, Variables>,
   initialOpts: UseClientRequestOptions<ResponseData, Variables> = {}
 ): [
   FetchData<ResponseData, Variables, TGraphQLError>,
   UseClientRequestResult<ResponseData, TGraphQLError>,
   ResetFunction
 ] {
-  if (typeof query !== 'string') {
-    throw new Error(
-      'Your query must be a string. If you are using the `gql` template literal from graphql-tag, remove it from your query.'
-    )
-  }
+  const queryString = stringifyDocumentNode(query)
+  const operationName = extractOperationName(query) ?? initialOpts.operationName
```

Comment:
is this logic correct? if you provide an operation name explicitly, shouldn't it have more priority than the one we infer from the query? unless I'm reading this wrong

Reply:
Corrected this.